### ISSUE TITLE
context.failure.message

### ISSUE THREAD

**Issue by jaxxon33** [Azure-Samples/active-directory-b2c-dotnetcore-webapp#20]
System.InvalidOperationException occurred
  HResult=0x80131509
  Message=Invalid non-ASCII or control character in header: 0x000A
  Source=<Cannot evaluate the exception source>
  StackTrace:
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.FrameHeaders.ThrowInvalidHeaderCharacter(Char ch)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.FrameHeaders.ValidateHeaderCharacters(String headerCharacters)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.FrameHeaders.ValidateHeaderCharacters(StringValues headerValues)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.FrameResponseHeaders.SetValueFast(String key, StringValues value)
   at Microsoft.AspNetCore.Server.Kestrel.Internal.Http.FrameHeaders.Microsoft.AspNetCore.Http.IHeaderDictionary.set_Item(String key, StringValues value)
   at Microsoft.AspNetCore.Http.Internal.DefaultHttpResponse.Redirect(String location, Boolean permanent)
   at WebApp_OpenIDConnect_DotNet.OpenIdConnectOptionsSetup.OnRemoteFailure(FailureContext context) in C:\Users\paul\Dropbox\Visual Studio\Tests\active-directory-b2c-dotnetcore-webapp-master\WebApp-OpenIDConnect-DotNet\OpenIdConnectOptionsSetup.cs:line 67
   at Microsoft.AspNetCore.Authentication.RemoteAuthenticationHandler`1.<HandleRemoteCallbackAsync>d__6.MoveNext()

"IDX10223: Lifetime validation failed. The token is expired.\nValidTo: '05/03/2017 03:12:15'\nCurrent time: '05/03/2017 13:23:11'."

**Comment by gsacavdm** [comment]
@jaxxon33 , can you provide more details. Was this just on a simple clone and run? Or was there some scenario that triggered it? The error seems to indicate there was a long running session involved... can you confirm?

**Comment by jaxxon33** [comment]
I just closed my browser, downloaded the zip, extracted then run. I was able to swap out the demo tenant for my own tenant which worked fine. I reproduced the error to confirm and assumed it was to do with the demo tenant.

Just now, before I replied, I tried to reproduce the problem by going through the exact process again, but this time it works perfectly.

Thanks for the sample and your prompt reply.
